src/yadmark/benchmark/external/apfelpy_utils.py

```diff
@@ -0,0 +1,312 @@
+"""
+    Utilities to help run APFEL++ benchmarks.
+"""
+import numpy as np
+
+# Q2 knots specs
+NQ = 50
+QMin = 1
+QMax = 200
+
+# Map observables names to APFEL++ methods
+MAP_HEAVINESS = {
+    "charm": 4,
+    "bottom": 5,
+    "top": 6,
+    "light": 0,  # TODO: Check combination
+    "total": 0,
+}
+
+
+def couplings(ap, pids, proc_type, obs_name):
+    """Return the corresponding coupling given a process type
+    and an observable.
+
+    Parameters
+    ----------
+    pids: int
+        PDG ID of the projectile
+    proc_type:
+        type of the DIS process
+    obs_name: str
+        name of the Yadism observable
+
+    Returns
+    -------
+    callable
+        a callable function that computes the coupling as a
+        function of the scale Q
+    """
+
+    # Effective charges
+    def _fBq(Q):
+        if proc_type == "EM":
+            # For Q=0 we only have electric charges
+            return ap.utilities.ElectroWeakCharges(0, False, pids)
+        return ap.utilities.ElectroWeakCharges(Q, False, pids)
+
+    def _fDq(Q):
+        if proc_type == "EM":
+            return [0.0]
+        return ap.utilities.ParityViolatingElectroWeakCharges(Q, False, pids)
+
+    # CKM matrix elements
+    def _fCKM(Q):
+        return ap.constants.CKM2
+
+    if proc_type == "CC":
+        coupling = _fCKM
+        if obs_name.startswith("g"):
+            raise ValueError("Yadism cannot compute polarised CC yet.")
+    else:
+        coupling = _fBq
+        if "F3" in obs_name or "gL" in obs_name or "g4" in obs_name:
+            coupling = _fDq
+
+    return coupling
+
+
+def tabulate_ncc(ap, obs_name, sfobj, nq, qmin, qmax, thrs):
+    """Tabulate the NC/EM structure function predictions.
+
+    Parameters
+    ----------
+    obs_name: str
+        name of the observable
+    sfobj: ap.init.InitalizeSFNCOjbectsZM
+        SF objects in Zero-Mass Flavour Number Scheme
+    nq: int
+        number of Q points
+    QMin: float
+        minimal value of Q
+    QMax: float
+        maximal value of Q
+    thrs: list
+        list of quark masses and thresholds
+
+    Returns
+    -------
+    callable:
+        Apfel++ callable functions to evaluate SF from
+    """
+    if "total" in obs_name:
+        tab_sf = ap.TabulateObjectD(sfobj[0].Evaluate, nq, qmin, qmax, 3, thrs)
+    elif "light" in obs_name:
+        tab_sf = ap.TabulateObjectD(
+            lambda Q: sfobj[1].Evaluate(Q)
+            + sfobj[2].Evaluate(Q)
+            + sfobj[3].Evaluate(Q),
+            nq,
+            qmin,
+            qmax,
+            3,
+            thrs,
+        )
+    elif "charm" in obs_name:
+        tab_sf = ap.TabulateObjectD(sfobj[4].Evaluate, nq, qmin, qmax, 3, thrs)
+    elif "bottom" in obs_name:
+        tab_sf = ap.TabulateObjectD(sfobj[5].Evaluate, nq, qmin, qmax, 3, thrs)
+    return tab_sf
+
+
+def tabulate_nc(ap, obs_name, sfobj, nq, qmin, qmax, thrs):
```

Comment:
```suggestion
def tabulate_cc(ap, obs_name, sfobj, nq, qmin, qmax, thrs):
```

Reply:
Ah, well spotted! This should indeed be `_cc`.